src/shipit_code_coverage/shipit_code_coverage/notifier.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from urllib.error import HTTPError
+import requests
+
+from cli_common.log import get_logger
+from cli_common.taskcluster import get_service
+
+from shipit_code_coverage.utils import wait_until
+
+
+logger = get_logger(__name__)
+
+
+class Notifier(object):
+
+    def __init__(self, revision, emails, client_id, access_token):
+        self.revision = revision
+        self.emails = emails
+        self.notify_service = get_service('notify', client_id, access_token)
+
+    def get_coverage_summary(self, changeset):
+        r = requests.get('https://uplift.shipit.staging.mozilla-releng.net/coverage/changeset_summary/%s' % changeset)
+        r.raise_for_status()
+
+        if r.status_code == 202:
+            return None
+
+        return r.json()
+
+    def notify(self):
+        content = ''
+
+        # Get pushlog and ask the backend to generate the coverage by changeset
+        # data, which will be cached.
+        r = requests.get('https://hg.mozilla.org/mozilla-central/json-pushes?changeset=%s&version=2&full' % self.revision)
+        r.raise_for_status()
+        push_data = r.json()
+        changesets = sum((data['changesets'] for data in push_data['pushes'].values()), [])
+
+        for changeset in changesets:
+            if any(text in changeset['desc'] for text in ['r=merge', 'a=merge']):
+                continue
+
+            try:
+                rev = changeset['node']
+                coverage = wait_until(lambda: self.get_coverage_summary(rev), 10)
+                if coverage is None:
+                    continue
+
+                if coverage['commit_covered'] < 0.2 * coverage['commit_added']:
+                    content += '* [%s](https://firefox-code-coverage.herokuapp.com/#/changeset/%s): %d covered out of %d added.\n' % (rev, rev, coverage['commit_covered'], coverage['commit_added'])  # noqa
+            except HTTPError as e:
+                continue
+
+        if content == '':
+            return
+        elif len(content) > 102400:
+            # Content is 102400 chars max
+            content = content[:102000] + '\n\n... Content max limit reached!'
```

Comment:
👍

Reply:
Copied from your static analysis reporter ;)
We should probably move some of the code in a common directory in the future.